crates/cext/src/transpiler/transpile_function.rs

```diff
@@ -198,6 +198,179 @@ pub unsafe extern "C" fn qk_transpile_stage_init(
     }
 }
 
+/// @ingroup QkTranspiler
+/// Run the preset layout stage of the transpiler on a circuit
+///
+/// The Qiskit transpiler is a quantum circuit compiler that rewrites a given
+/// input circuit to match the constraints of a QPU and optimizes the circuit
+/// for execution. This function runs the second stage of the preset pass manager
+/// **layout**, which chooses the initial mapping of virtual qubits to
+/// physical qubits, including expansion of the circuit to contain explicit
+/// ancillas. You can refer to
+/// @verbatim embed:rst:inline :ref:`transpiler-preset-stage-layout` @endverbatim for more details.
+///
+/// This function should only be used with circuits constructed
+/// using Qiskit's C API. It makes assumptions on the circuit only using features exposed via C,
+/// if you are in a mixed Python and C environment it is typically better to invoke the transpiler
+/// via Python.
+///
+/// This function is multithreaded internally and will launch a thread pool
+/// with threads equal to the number of CPUs reported by the operating system by default.
+/// This will include logical cores on CPUs with simultaneous multithreading. You can tune the
+/// number of threads with the ``RAYON_NUM_THREADS`` environment variable. For example, setting
+/// ``RAYON_NUM_THREADS=4`` would limit the thread pool to 4 threads.
+///
+/// @param dag A pointer to the circuit to run the transpiler on.
+/// @param target A pointer to the target to compile the circuit for.
+/// @param options A pointer to an options object that defines user options. If this is a null
+///   pointer the default values will be used. See ``qk_transpile_default_options``
+///   for more details on the default values.
+/// @param layout A pointer to a pointer to a ``QkTranspileLayout`` object. On a successful
+///   execution (return code 0) a pointer to the layout object created transpiler will be written
+///   to this pointer. The inner pointer for this can be null if there is no existing layout
+///   object. Typically if you run `qk_transpile_stage_init` you would take the output layout from
+///   that function and use this as the input for this. But if you don't have a layout the inner
+///   pointer can be null and a new `QkTranspileLayout` will be allocated and that pointer will be
+///   set for the inner value of the layout here.
+/// @param error A pointer to a pointer with an nul terminated string with an error description.
+///   If the transpiler fails a pointer to the string with the error description will be written
+///   to this pointer. That pointer needs to be freed with ``qk_str_free```. This can be a null
```

Comment:
Just to help solve some CI issues.
```suggestion
///   to this pointer. That pointer needs to be freed with ``qk_str_free``. This can be a null
```